pkg/stub/handler.go

```diff
@@ -1138,3 +1106,116 @@ func GetNamespace() string {
+}
+
+func (h *Handler) sufficientSteadyStateForAnOperator(cfg *v1.Config) bool {
+	return util.ConditionTrue(cfg, v1.SamplesExist) &&
+		util.ConditionFalse(cfg, v1.ImageChangesInProgress) &&
+		h.version == cfg.Status.Version
+}
+
+func (h *Handler) areStreamsAtCorrectVersion() bool {
+	for key, _ := range h.imagestreamFile {
+		// remember, our get wrapper goes through the lister, so we are only hitting the controller cache
+		skipped, ok := h.skippedImagestreams[key]
+		if skipped && ok {
+			continue
+		}
+		is, err := h.imageclientwrapper.Get(key)
+		if err != nil {
+			if !h.upsertInProgress {
+				// not all imagestreams may be created yet during initial or bulk creations
+				logrus.Warningf("areStreamsAtCorrectVersion: get of stream %s got error: %s", key, err.Error())
+			}
+			return false
+		}
+		if is == nil {
+			if !h.upsertInProgress {
+				// not all imagestreams may be created yet during initial or bulk creations
+				logrus.Warningf("areStreamsAtCorrectVersion: got nil for stream %s but no error", key)
+			}
+			return false
+		}
+		if is.Annotations == nil {
+			logrus.Warningf("areStreamsAtCorrectVersion: stream %s has no annotation map", key)
+			return false
+		}
+		isv, ok := is.Annotations[v1.SamplesVersionAnnotation]
+		if !ok {
+			logrus.Warningf("areStreamsAtCorrectVersion: stream %s does not have version annotation", key)
+			return false
+		}
+		if isv != h.version {
+			logrus.Warningf("areStreamsAtCorrectVersion: stream %s is at version %s instead of %s", key, isv, h.version)
+			return false
+		}
+	}
+	logrus.Printf("areStreamsAtCorrectVersion: all streams are now at the correct version %s", h.version)
+	return true
+}
+
+func (h *Handler) numOfManagedImageStreams() int {
+	rc := len(h.imagestreamFile) - len(h.skippedImagestreams)
+	return rc
+}
+
+func (h *Handler) numOfManagedImageStreamsPresent() int {
+	rc := 0
+	for key, _ := range h.imagestreamFile {
+		// remember, our get wrapper goes through the lister, so we are only hitting the controller cache
+		skipped, ok := h.skippedImagestreams[key]
+		if skipped && ok {
+			continue
+		}
+		is, err := h.imageclientwrapper.Get(key)
+		if err != nil {
+			if !h.upsertInProgress {
+				// not all imagestreams may be created yet during initial or bulk creations
+				logrus.Warningf("numOfManagedImageStreamsPresent: get of stream %s got error: %s", key, err.Error())
+			}
+			continue
+		}
+		if is == nil {
+			if !h.upsertInProgress {
+				// not all imagestreams may be created yet during initial or bulk creations
+				logrus.Warningf("numOfManagedImageStreamsPresent: got nil for stream %s but no error", key)
+			}
+			continue
+		}
+		if is.Labels == nil {
+			logrus.Warningf("numOfManagedImageStreamsPresent: stream %s has no labels map", key)
+			continue
+		}
+		managed, ok := is.Labels[v1.SamplesManagedLabel]
+		if !ok {
+			logrus.Warningf("numOfManagedImageStreamsPresent: stream %s does not have managed label", key)
+			continue
+		}
+		if managed != "true" {
```

Comment:
Why is managed a string instead of a boolean? Wouldn't that simplify things some?

Reply:
label values can only be strings, so no, type conversions here just add stuff